### DISCUSSION TITLE
Custom script - filter changelob between dates

### DISCUSSION THREAD

**Discussion by ziggekatten** [netbox-community/netbox#15565] [Help Wanted!] [answered]
I need to do some large scale fix, and need to get the data from the changelog between specfic time points and programatically revert some changes,

In effect: I would like to do an ObjectChange.objects.filter() with a date greater that x, and less than y. I can't for my life figure out how to do that.

**Comment by zmaster7** [comment, marked as answer]
Your variables to filter on need to be in datetime format, and then you can filter on time__lte and time__gte.

Here's a quick example (filtered between 1/1/2020-present):

```
from datetime import datetime, timezone
from extras.models import ObjectChange

startDateTime = datetime(2020,1,1,0,0,0,tzinfo=datetime.utcnow().astimezone().tzinfo)
endDateTime = datetime.now(timezone.utc).astimezone()

Records = ObjectChange.objects.filter(time__gte=startDateTime,time__lte=endDateTime)
```
